### DISCUSSION TITLE
CE Install Won't Send Email via SMTP or SendGrid API

### DISCUSSION THREAD

**Discussion by johnnydanger** [plausible/analytics#4765] [Self-Hosted Support]
I've tried everything and am pulling my hair out.  New CE install, and no matter what I can't get it to send email.

Here's the log.  Very illuminating...

plausible-ce-plausible-1 | Loading plausible..
plausible-ce-plausible-1 | Starting dependencies..
plausible-ce-plausible-1 | Starting repos..
plausible-ce-plausible-1 | Plausible.Repo database already exists
plausible-ce-plausible-1 | Plausible.IngestRepo database already exists
plausible-ce-plausible-1 | Creation of Db successful!
plausible-ce-plausible-1 | Loading plausible..
plausible-ce-plausible-1 | Starting dependencies..
plausible-ce-plausible-1 | Starting repos..
plausible-ce-plausible-1 | 01:06:01.395 [notice] Certificate for *REDACTED* is valid until 2025-01-30. Next renewal is scheduled for 2024-12-31.
plausible-ce-plausible-1 | 01:06:11.771 request_id=GAQBSVPp9Sh6ZGYAACOB [error] Failed to send e-mail

"Failed to send e-mail"?  No kidding!

I tried turning on LOG_LEVEL=debug and it shows the email assembled, but there's no further details about why it's failing to send.

I have tried telnet from the server and both 25 and 587 work fine and I can manually get the AUTH to work using the info I've put into the .env so that's not the issue.  I have tried both Bamboo.Mua and Bamboo.SendGridAdapter, as well as SMTP settings for SendGrid and the SendGrid API Key.

Help!  Here's my current .env file:

MAILER_ADAPTER=Bamboo.Mua
MAILER_EMAIL=do-not-reply@plausible.myserver.cloud
MAILER_NAME=Plausible
SMTP_HOST_ADDR=smtp.sendgrid.net
SMTP_HOST_PORT=587
SMTP_USER_NAME=apikey
SMTP_USER_PWD=SG.redacted
SMTP_HOST_SSL_ENABLED=false
SENDGRID_API_KEY=SG.redacted

**Comment by johnnydanger** [comment]
And here's a debug log of the generated email:

```
plausible-ce-plausible-1 | %Bamboo.Email{from: {"Plausible", "do-not-reply@plausible.*REDACTED*"}, to: [nil: "*MY EMAIL ADDRESS REDACTED*"], cc: [], bcc: [], subject: "[Plausible CE] You've been invited to justabigclock.com", html_body: "Hey,\n<br /><br />\n*REDACTED* has invited you to join the justabigclock.com site on Plausible CE.\n<a href=\"https://*REDACTED*/register/invitation/DdWoKw4WCIWcscgWoaDvt\">Click here</a> to create your account. The link is valid for 48 hours after this email is sent.\n<br /><br />\nPlausible is a lightweight and open-source website analytics tool. We hope you like our simple and ethical approach to tracking website visitors.\n\n<br /><br />\n\n--\n<br /><br />\n<a href=\"https://*REDACTED*\">https://*REDACTED*</a><br />\n", text_body: nil, headers: %{}, attachments: [], assigns: %{invitation: %Plausible.Auth.Invitation{__meta__: #Ecto.Schema.Metadata<:loaded, "invitations">, id: 29, invitation_id: "DdWoKw4WCIWcscgWoaDvt", email: "*REDACTED*", role: :admin, inviter_id: 1, inviter: %Plausible.Auth.User{__meta__: #Ecto.Schema.Metadata<:loaded, "users">, id: 1, email: "*REDACTED*", password_hash: "*REDACTED*", old_password: nil, password: nil, password_confirmation: nil, name: "*REDACTED*", last_seen: ~N[2024-11-01 22:47:13], trial_expiry_date: ~D[2124-11-01], theme: :system, email_verified: true, previous_email: nil, accept_traffic_until: ~D[2124-11-15], notes: nil, allow_next_upgrade_override: false, totp_enabled: false, totp_secret: nil, totp_token: nil, totp_last_used_at: nil, grace_period: nil, sessions: #Ecto.Association.NotLoaded<association :sessions is not loaded>, site_memberships: #Ecto.Association.NotLoaded<association :site_memberships is not loaded>, sites: #Ecto.Association.NotLoaded<association :sites is not loaded>, api_keys: #Ecto.Association.NotLoaded<association :api_keys is not loaded>, google_auth: #Ecto.Association.NotLoaded<association :google_auth is not loaded>, subscription: #Ecto.Association.NotLoaded<association :subscription is not loaded>, enterprise_plan: #Ecto.Association.NotLoaded<association :enterprise_plan is not loaded>, inserted_at: ~N[2024-11-01 22:47:13], updated_at: ~N[2024-11-01 22:47:13]}, site_id: 1, site: %Plausible.Site{__meta__: #Ecto.Schema.Metadata<:loaded, "sites">, id: 1, domain: "justabigclock.com", timezone: "America/Los_Angeles", public: false, locked: false, stats_start_date: ~D[2024-11-01], native_stats_start_at: ~N[2024-11-01 22:47:38], allowed_event_props: nil, conversions_enabled: true, props_enabled: true, funnels_enabled: true, ingest_rate_limit_scale_seconds: 60, ingest_rate_limit_threshold: nil, domain_changed_from: nil, domain_changed_at: nil, imported_data: nil, installation_meta: %Plausible.Site.InstallationMeta{id: "*REDACTED*", installation_type: nil, script_config: %{"404" => false, "file-downloads" => false, "hash" => false, "outbound-links" => false, "pageview-props" => false, "revenue" => false, "tagged-events" => false}}, members: #Ecto.Association.NotLoaded<association :members is not loaded>, memberships: #Ecto.Association.NotLoaded<association :memberships is not loaded>, invitations: #Ecto.Association.NotLoaded<association :invitations is not loaded>, goals: #Ecto.Association.NotLoaded<association :goals is not loaded>, revenue_goals: #Ecto.Association.NotLoaded<association :revenue_goals is not loaded>, google_auth: #Ecto.Association.NotLoaded<association :google_auth is not loaded>, weekly_report: #Ecto.Association.NotLoaded<association :weekly_report is not loaded>, monthly_report: #Ecto.Association.NotLoaded<association :monthly_report is not loaded>, ownership: #Ecto.Association.NotLoaded<association :ownership is not loaded>, owner: #Ecto.Association.NotLoaded<association :owner is not loaded>, from_cache?: false, entry_type: nil, pinned_at: nil, import_data_loaded: false, earliest_import_start_date: nil, latest_import_end_date: nil, complete_import_ids: [], inserted_at: ~N[2024-11-01 22:47:38], updated_at: ~N[2024-11-01 22:50:32]}, inserted_at: ~N[2024-11-02 00:49:39], updated_at: ~N[2024-11-02 00:49:39]}}, private: %{tag: "new-user-invitation", message_params: %{"MessageStream" => "priority", "TrackOpens" => false}, html_layout: {PlausibleWeb.LayoutView, "priority_email.html"}, text_layout: false, view_module: PlausibleWeb.EmailView, view_template: "new_user_invitation.html"}, blocked: false}
plausible-ce-plausible-1 | 
plausible-ce-plausible-1 | 00:49:40.499 request_id=GAQAYoUHTQsjE9kAACNx [error] Failed to send e-mail
plausible-ce-plausible-1 | 00:49:40.501 request_id=GAQAYoUHTQsjE9kAACNx [debug] QUERY OK db=1.4ms
```

**Comment by ruslandoga** [comment]
👋 @johnnydanger 

I'll open a PR to improve the error message, but in the meanwhile let's get the full error by sending the email manually.

```console
$ cd plausible-ce
$ docker compose exec plausible bin/plausible remote
```

```elixir
iex> Application.get_env(:plausible, Plausible.Mailer)
# just a sanity check, what does it return? note that it would show some secrets, they can be redacted :)

iex> you = "you@gmail.com"
iex> email = Bamboo.Email.new_email(to: you, from: PlausibleWeb.Email.mailer_email_from(), text_body: "test")
iex> Plausible.Mailer.deliver_now(email)
# what does it return?
```

You can try it both with SMTP and SendGrid adapters. Note that the SendGrid adapter doesn't use SMTP but HTTP.

**Comment by johnnydanger** [comment]
Well I'll be pickled!  You're a genius. I know where to go to try and resolve the issue now.  Sender Identity 🤦🏻‍♂️

Thanks so much for the quick response.  I'm confident I can solve this post haste.


====


iex(plausible@86272997cf0d)3>  email = Bamboo.Email.new_email(to: you, from: PlausibleWeb.Email.mailer_email_from(), text_body: "test")
%Bamboo.Email{
  from: {"Plausible", "do-not-reply@plausible.*REDACTED*"},
  to: "you@gmail.com",
  cc: nil,
  bcc: nil,
  subject: nil,
  html_body: nil,
  text_body: "test",
  headers: %{},
  attachments: [],
  assigns: %{},
  private: %{},
  blocked: false
}
iex(plausible@86272997cf0d)4> Plausible.Mailer.deliver_now(email)
{:error,
 %Mua.SMTPError{
   code: 550,
   lines: ["550 The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity requirements\r\n"]
 }}

**Reply by johnnydanger** [reply]
Problem solved.  Thanks again.

Adding the Bamboo errors to the Plausible log would be super handy to increase visibility of potential mailer issues.  Much appreciated.

Cheers.
